cmd/ip-masq-agent/ip-masq-agent.go

```diff
@@ -370,8 +390,13 @@ func writeNonMasqRule(lines *bytes.Buffer, cidr string) {
 
 const masqRuleComment = `-m comment --comment "ip-masq-agent: outbound traffic is subject to MASQUERADE (must be last in chain)"`
 
-func writeMasqRule(lines *bytes.Buffer) {
-	writeRule(lines, utiliptables.Append, masqChain, masqRuleComment, "-j", "MASQUERADE")
+func writeMasqRule(iptablesVersion string, lines *bytes.Buffer) {
+	if semver.Compare(iptablesVersion, iptablesMinVersion) >= 0 {
+		writeRule(lines, utiliptables.Append, masqChain, "-m", "addrtype", "--src-type", "LOCAL", "-j", "RETURN")
```

Comment:
Can you explain more about why we are doing an early return here?

Reply:
The reason for the early return is so that local addresses aren't masqueraded. There were issues in our own testing without the early return.

Reply:
From my local iptables rules it seems there've been early returns for local traffic, could you share the rules in IP-MASQ-AGENT chain before/after your fix?

![image](https://user-images.githubusercontent.com/8267625/89980836-3ed3bd80-dca5-11ea-9ca7-eb90f8c5653b.png)

Reply:
Still not understanding what would we need an early return for local traffic here - the default rules should have included all the RFC1918 and reserved range as the non-masquerade destination already. Curious what kind of local test did you perform?